File: e2e/playwright-typescript/tests/translation/language-support.spec.ts
import { test } from '@/fixtures';
import { TaskContext } from '@/context/TaskContext';
import { TaskPriority } from '@/data/TaskPriority';
import { TaskStatus } from '@/data/TaskStatus';
import * as allure from "allure-js-commons";
import { AllureEpic } from '@/data/AllureEpic';

test.beforeAll(async () => {
  await allure.epic(AllureEpic.TRANSLATION);
  await allure.feature('Language support');
  await allure.tms('104');
});

test.describe('Language support', () => {
  let context: TaskContext;

  test.beforeEach(async ({ support }) => {
    context = new TaskContext({ status: TaskStatus.TODO, priority: TaskPriority.LOW });
    const response = context.createTaskResponse();

    await support.mock.api.getTasks([response]);
    await support.mock.api.createTask(response);
  });

  test('should switch UI to Spanish when ES is selected', async ({ step, validate, support }) => {
    // given
    await step.navigation.openMainPage();

    // when
    await step.language.selectLanguage('ES');

    // then
    await validate.language.uiIsInSpanish();
    await validate.language.statusTagShowsText('TODO', 'Por hacer');
    await validate.language.priorityTagShowsText('LOW', 'Baja');
  });
});

File: e2e/capybara-ruby/spec/translation/language_support_spec.rb
# frozen_string_literal: true

require 'spec_helper'

require_relative '../../context/task_context'
require_relative '../../data/allure_epic'
require_relative '../../data/task_priority'
require_relative '../../data/task_status'

RSpec.describe 'Language support', type: :system,
                                   epic: AllureEpic::TRANSLATION,
                                   feature: 'Language support',
                                   tms: '104' do
  let(:context) do
    TaskContext.new(status: TaskStatus::TODO, priority: TaskPriority::LOW)
  end

  before do
    response = context.create_task_response
    support.mock.api
         .get_tasks([response])
         .create_task(response)
  end

  it 'should switch UI to Spanish when ES is selected' do
    # given
    step.navigation.open_main_page

    # when
    step.language.select_language('ES')

    # then
    validate.language.ui_is_in_spanish
    validate.language.status_tag_shows_text('TODO', 'Por hacer')
    validate.language.priority_tag_shows_text('LOW', 'Baja')
  end
end
